src/Agent.PluginHost/Program.cs

```diff
@@ -24,10 +24,16 @@ public static class Program
 
         public static int Main(string[] args)
         {
-            // We can't use the new SocketsHttpHandler for now for both Windows and Linux
+            // In .NET Core 2.1, we couldn't use the new SocketsHttpHandler for now for both Windows and Linux
             // On linux, Negotiate auth is not working if the TFS url is behind Https
             // On windows, Proxy is not working
-            AppContext.SetSwitch("System.Net.Http.UseSocketsHttpHandler", false);
+            // But on ARM/ARM64 Linux, the legacy curl dependency is problematic
+            // https://github.com/dotnet/runtime/issues/28891
+            if (!(PlatformUtil.RunningOnLinux && (PlatformUtil.IsArm || PlatformUtil.IsArm64)))
+            {
+                AppContext.SetSwitch("System.Net.Http.UseSocketsHttpHandler", false);
```

Comment:
#2875 will replace this ugly logic with a Knob (or similar)

Reply:
That's a cool idea, have a knob that has a runtime source but also create a platform source knob that can set a value based on which platform we are running on.